src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelperFactory.cs

```diff
@@ -32,7 +32,12 @@ public IUrlHelper GetUrlHelper(ActionContext context)
             }
 
             // Perf: Create only one UrlHelper per context
-            var urlHelper = httpContext.Items[typeof(IUrlHelper)] as IUrlHelper;
+            IUrlHelper urlHelper = null;
+
+            if (httpContext.Items.ContainsKey(typeof(IUrlHelper)))
+            {
+                urlHelper = httpContext.Items[typeof(IUrlHelper)] as IUrlHelper;
+            }
```

Comment:
Also this block needs some whitespace to make it clear that it's not related to the other block.